core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -126,24 +126,21 @@ oppia.factory('ExtractImageFilenamesFromStateService', [
         var filepathObject = JSON.parse(
           imageTagList[i].getAttribute('filepath-with-value'));
         // The images already there in Oppia have image filenames as the value
-        // for the attribute 'filepath-with-value'. In explorations the
-        // the attribute value is an object of the form --
+        // for the attribute 'filepath-with-value'. We later on want to have
```

Comment:
So, just to check -- with these changes, what exactly is being stored in the backend for explorations? The definition of an exploration's contents needs to be really solid, and we need to be very clear on what it is. In particular, the filepaths are still strings, right? Is the "filepath object" you're creating here ever written to the backend? (It should be a domain object, btw.)

If you want to universally change the format of all image RTE component specifications then you need to do an exploration migration for this and have both the backend and frontend code support the new format _only_. It's necessary to think about the explorations in production and ensure that they remain safe, while keeping their format consistent with newly-created explorations.

So, i guess my question is: does the current code in develop cause anything to be changed in what gets stored in Exploration domain objects in the frontend (including the format of the RTE HTML for image components)? If not, then this code is fine. If so then further modifications are needed and we might need to talk through this.

Reply:
Yes, the filepaths are still strings. No the filepath-object that I am creating here is never written to the backend. The purpose of it is to just get the dimensions and filenames all together in an object. The dimensions are required for the image loading indicator. We are creating this object with a default dimensions for every image, since we don't have the dimensions of the images stored in the backend as of now.

Yes, that exploration migration that would change the filepath value is scheduled for the next milestone.
The code in the develop branch, stores the filepath value as object. But this current PR is for reverting that change. Once this PR get's in the develop, all the backend and frontend will support the format that already existed (filepath as string).

For the future (The next milestone and next release (Aug) ) ---
The exploration migration will be run and PR will be created that stores the value of the filepath as an object.

I hope this clears it all.
Thanks

Reply:
Thanks for the explanation @ishucr7. That's useful to know!

Please make sure to revert all instances of "filepath object" throughout the codebase when what is stored is actually a filename -- it's important that we get this right.